app/shared/modals/reservation-rate-report/ReservationsRateReport.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component, PropTypes } from 'react';
+import moment from 'moment';
+import FontAwesome from 'react-fontawesome';
+import Button from 'react-bootstrap/lib/Button';
+import Modal from 'react-bootstrap/lib/Modal';
+import Row from 'react-bootstrap/lib/Row';
+import FormGroup from 'react-bootstrap/lib/FormGroup';
+import ControlLabel from 'react-bootstrap/lib/ControlLabel';
+import { Typeahead } from 'react-bootstrap-typeahead';
+
+import DatePicker from 'shared/date-picker';
+import DateTimeRange from 'shared/form-fields/DateTimeRange';
+import 'react-bootstrap-typeahead/css/Typeahead.css';
+import 'react-bootstrap-typeahead/css/Typeahead-bs4.css';
+
+
+const initialState = {
+  unitSelections: [],
+  dateStart: moment().subtract(30, 'days').format('YYYY-MM-DD'),
+  dateEnd: moment().format('YYYY-MM-DD'),
+  times: {
+    begin: { time: '08:00' },
+    end: { time: '16:00' },
+  },
+  loading: false,
+};
+
+export default class ReservationsRateReportModal extends Component {
+  static propTypes = {
+    onHide: PropTypes.func.isRequired,
+    show: PropTypes.bool.isRequired,
+    units: PropTypes.object.isRequired,
+    fetchReservationsRateReport: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = initialState;
+    this.handleChange = this.handleDateChange.bind(this);
+    this.handleTimeChange = this.handleTimeChange.bind(this);
+    this.downloadReport = this.downloadReport.bind(this);
+  }
+
+  handleDateChange(date) {
+    this.setState({ ...date });
+  }
+
+  handleTimeChange(time) {
+    this.setState({ times: time });
+  }
+
+  downloadReport() {
+    this.setState({ loading: true });
+    this.props.fetchReservationsRateReport({
+      start_date: this.state.dateStart,
+      end_date: this.state.dateEnd,
+      start_time: this.state.times.begin.time,
+      end_time: this.state.times.end.time,
+      units: this.state.unitSelections.map(selection => selection.value),
+    }).then(() => {
+      this.setState({ loading: false });
+    }).catch(() => {
+      this.setState({ loading: false });
```

Comment:
There is no error handling here, should there be? Should the user get some kind of message/notification if things did not go as planned?

Reply:
Most errors are related to user not selecting correct data in the modal, such as not selecting any units or setting a start date that is after the end date. These are handled on the back end right now though. So should I just implement some local form errors, or just somehow display the error message from the back end?